x/programs/rust/wasmlanche_sdk/src/program.rs

```diff
@@ -8,8 +8,8 @@ use crate::{errors::StateError, host::call_program, state::State};
 pub struct Program([u8; Self::LEN]);
 
 impl Program {
-    /// The length of ids.ID
-    pub const LEN: usize = 32;
```

Comment:
isn't 32 bytes still needed here the id() method will return an ID that should be 32 bytes right?

Reply:
I was under the impression that programs id's would also be saved as `Address` with a different typeID

Reply:
I guess we should verify how that will look. HyperShift still used a 32bit `ID`[1]. So if we are returning a program.id() that should be 32bits. I think we can achieve what your describing by dropping the first bit. 


[1] https://github.com/ava-labs/hypersdk/blob/ad3ac9427c178c85cc55e5dbc9b044b04152d89e/examples/morpheusvm/storage/storage.go#L74

https://github.com/ava-labs/hypersdk/blob/b3da5dec37df3f6bb7a27069e6d878f485aa4b34/x/programs/rust/wasmlanche_sdk/src/program.rs#L14C1-L18C6

Reply:
ah ok, guess we'll keep it as id then. Can see this also becoming generic tho